test/e2e_node/util.go

```diff
@@ -641,20 +640,8 @@ func WaitForPodInitContainerToFail(ctx context.Context, c clientset.Interface, n
 }
 
 func nodeNameOrIP() string {
-	// Check if the node name in test context can be resolved
-	if ips, err := net.LookupIP(framework.TestContext.NodeName); err != nil {
-		if dnsErr, ok := err.(*net.DNSError); ok && dnsErr.IsNotFound {
-			// if it can't be resolved, pick a host interface
-			if ip, err := utilnet.ChooseHostInterface(); err == nil {
-				return ip.String()
-			}
-		}
-	} else {
-		if len(ips) > 0 {
-			// yay, node name resolved correctly, pick the first
-			return ips[0].String()
-		}
-	}
-	// fallback to node name in test context
-	return framework.TestContext.NodeName
+	return "localhost"
+}
+
+func metricsURL(port int, path string) string {
```

Comment:
did you try to unify the logic of creating the URL?

Reply:
this needs a return

Reply:
whoops, didn't mean to push this